src/views/Main/Schemas/Components/PropertyType.js

```diff
@@ -6,40 +6,7 @@ import styles from '../styles.module.css';
 
 export class PropertyType extends React.Component {
   static propTypes = {
-    propertyType: PropTypes.object,
-    propertyTypePage: PropTypes.bool,
-    error: PropTypes.func,
-    updateFn: PropTypes.func,
-    schemaName: PropTypes.string,
-    schemaNamespace: PropTypes.string
-  }
-
-  static contextTypes = {
-    isAdmin: PropTypes.bool
-  }
-
-  deleteProp = () => {
-    const { schemaName, schemaNamespace, propertyType, updateFn, error } = this.props;
-    EntityDataModelApi.removePropertyTypesFromSchema(Utils.getFqnObj(schemaNamespace, schemaName), [propertyType])
-    .then(() => {
-      return updateFn();
-    }).catch(() => {
-      return error();
-    });
-  }
-
-  renderDeleteButton = () => {
-    const className = (this.props.propertyTypePage) ? styles.hidden : StringConsts.EMPTY;
-    if (this.context.isAdmin) {
-      return (
-        <td className={className}>
-          <button className={styles.deleteButton} onClick={this.deleteProp}>-</button>
-        </td>
-      );
-    }
-    return (
-      <td className={className} />
-    );
+    propertyType: PropTypes.object
```

Comment:
Check out `components/propertytype/PropertyTypeStorage.js`. There's a pre-build `PropType` for `PropertyType`s that you can drop into your `propTypes` dict :)